src/cloud-api-adaptor/test/provisioner/kustomize.go

```diff
@@ -170,6 +170,57 @@ func (kh *KustomizeOverlay) SetKustomizeSecretGeneratorLiteral(secretName string
 	return nil
 }
 
+// SetKustomizeSecretGeneratorEnvs updates the kustomization YAML by adding the `env` on the
```

Comment:
I'm wondering if you really need this (and the env file) or you can simply update the peer-pods-cm with the secret and access_key ?

Reply:
hmm.. that's a good question. Probably I could have updated peer-pods-cm to inject the credentials as literals, but I would still need to create a dummy aws-cred.env file otherwise kustomize fails to load the kustomization.yaml. In the end, writing the credentials to the file and using this  `SetKustomizeSecretGeneratorEnv` is the least of the problems. Also writing to aws-cred.env is an extra road-guard to prevent publishing the credentials on github by mistake (yes, I did it once :(